File: projeto_integrador_backend/src/main/java/com/projetointegrador/model/Propriedade.java
package com.projetointegrador.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;

@Entity
@Table(name = "propriedade")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Propriedade {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_propriedade")
    private Long idPropriedade;

    @Column(name = "nome", nullable = false)
    private String nome;

    @Column(name = "localidade")
    private String localidade;

    @Column(name = "cidade")
    private String cidade;

    @Column(name = "telefone")
    private String telefone;

    @Column(name = "area_total", precision = 10, scale = 2)
    private BigDecimal areaTotal;

    @Column(name = "latitude", precision = 10, scale = 7)
    private BigDecimal latitude;

    @Column(name = "longitude", precision = 10, scale = 7)
    private BigDecimal longitude;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "id_usuario", nullable = false)
    private Usuario usuario;

    @Column(name = "data_criacao", nullable = false, updatable = false)
    private java.time.LocalDateTime dataCriacao;

    @Column(name = "data_atualizacao")
    private java.time.LocalDateTime dataAtualizacao;

    @PrePersist
    protected void onCreate() {
        dataCriacao = java.time.LocalDateTime.now();
        dataAtualizacao = java.time.LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        dataAtualizacao = java.time.LocalDateTime.now();
    }
}

File: projeto_integrador_flutter/lib/models/propriedade.dart
import 'package:json_annotation/json_annotation.dart';

part 'propriedade.g.dart';

@JsonSerializable()
class Propriedade {
  @JsonKey(name: 'idPropriedade')
  final int? idPropriedade;
  
  final String nome;
  final String? localidade;
  final String? cidade;
  final String? telefone;
  
  @JsonKey(name: 'areaTotal')
  final double? areaTotal;
  
  final double? latitude;
  final double? longitude;
  
  @JsonKey(name: 'idUsuario')
  final int idUsuario;

  Propriedade({
    this.idPropriedade,
    required this.nome,
    this.localidade,
    this.cidade,
    this.telefone,
    this.areaTotal,
    this.latitude,
    this.longitude,
    required this.idUsuario,
  });

  factory Propriedade.fromJson(Map<String, dynamic> json) =>
      _$PropriedadeFromJson(json);

  Map<String, dynamic> toJson() => _$PropriedadeToJson(this);

  Propriedade copyWith({
    int? idPropriedade,
    String? nome,
    String? localidade,
    String? cidade,
    String? telefone,
    double? areaTotal,
    double? latitude,
    double? longitude,
    int? idUsuario,
  }) {
    return Propriedade(
      idPropriedade: idPropriedade ?? this.idPropriedade,
      nome: nome ?? this.nome,
      localidade: localidade ?? this.localidade,
      cidade: cidade ?? this.cidade,
      telefone: telefone ?? this.telefone,
      areaTotal: areaTotal ?? this.areaTotal,
      latitude: latitude ?? this.latitude,
      longitude: longitude ?? this.longitude,
      idUsuario: idUsuario ?? this.idUsuario,
    );
  }
}
